File: derrick/detectors/general/derrick.py
#! /usr/bin/env python
# -*- coding: utf-8 -*-

from __future__ import absolute_import, division, print_function

import os

from derrick.core.detector import Detector
from derrick.core.common import DERRICK_VERSION


def normalize_name(name):
    # return re.sub(r'[^a-z0-9]', '', name.lower())
    return name


def get_project_name():
    project = os.path.basename(os.path.abspath(""))
    if project:
        return normalize_name(project)

    return 'default'


class DerrickDetector(Detector):
    def execute(self, *args, **kwargs):
        return {
            "derrick_version": DERRICK_VERSION,
            "project_name": get_project_name()
        }

File: detectors/general/derrick.go
package general

import (
	"path/filepath"

	"github.com/cloud-native-application/derrick-go/common"
)

type DerrickDetector struct {
}

func getProjectName() (string, error) {
	p, err := filepath.Abs(".")
	if err != nil {
		return "", err
	}
	return filepath.Base(p), nil
}

func (detector DerrickDetector) Execute() (map[string]string, error) {
	var projectName = "default"
	base, err := getProjectName()
	if err != nil {
		return nil, err
	}
	if base != "" {
		projectName = base
	}
	return map[string]string{
		common.DerrickVersion: common.DERRICK_VERSION,
		common.ProjectName:    projectName,
	}, nil

}

func (detector DerrickDetector) Name() string {
	return "DerrickDetector"
}
